dxr/plugins/xbl/analyzer.py

```diff
@@ -0,0 +1,91 @@
+from warnings import warn
+import xml.parsers.expat as expat
+
+from dxr.indexers import Extent, Position
+from dxr.plugins.xbl.refs import PLUGIN_NAME, TypeRef
+
+
+class XBLAnalyzer(object):
+    def __init__(self, path, tree, contents, encoding):
+        self.tree = tree
+        self.path = path
+        self.contents = contents
+        self.lines = contents.splitlines(True)
+        self.parser = expat.ParserCreate(encoding)
+        # Stack of namespaces, issued by the 'id' attr of bindings.
+        self.namespace = []
+        self.needles = []
+        self.refs = []
+        # Late bind our custom handlers to the parser
+        self.parser.StartElementHandler = self.StartElementHandler
+        self.parser.EndElementHandler = self.EndElementHandler
+        try:
+            self.parser.Parse(self.contents.encode(encoding))
+        except expat.ExpatError as err:
+            warn('Exception occurred on parsing XBL file {}'.format(self.path.encode('utf-8')))
+            warn(err.message)
+
+    def extent_for_name(self, name):
+        """Return Extent for the next occurrence of name."""
+        # Because the parser does not provide location info on attr positions,
+        # we can only search for them from the current position.
+        # First check the current line.
+        offset = self.lines[self.parser.CurrentLineNumber - 1].find(name, self.parser.CurrentColumnNumber - 1)
```

Comment:
Agreed, it's possible that the buggy situation could happen; I'll leave a comment and skipped test. Similarly the xpidl plugin [did the same](https://github.com/mozilla/dxr/blob/91cef85924e5ee456350e814e59fd6cfa50013f0/dxr/plugins/xpidl/visitor.py#L22). However in practice the circumstance should be rare, it would require a binding called "binding" or an interface named "implements."